### PR TITLE
Added equals() impl for MessageInfo

### PR DESCRIPTION
**Pull request by dharju**

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java [hunk 1]
```diff
@@ -166,6 +166,17 @@ public long getOperationTimeMs() {
     return operationTimeMs;
   }
 
+  @Override
+  public boolean equals(Object thatObject) {
+    if (thatObject instanceof MessageInfo) {
```
Comment (vgkholla) [inline ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java]: was this autogenerated by intellij?

File: ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java [hunk 2]
```diff
@@ -167,14 +168,23 @@ public long getOperationTimeMs() {
...
+      return true;
     }
-    return false;
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    MessageInfo that = (MessageInfo) o;
+    return size == that.size && expirationTimeInMs == that.expirationTimeInMs && isDeleted == that.isDeleted
+        && isTtlUpdated == that.isTtlUpdated && accountId == that.accountId && containerId == that.containerId
+        && operationTimeMs == that.operationTimeMs && Objects.equals(key, that.key) && Objects.equals(crc, that.crc);
```
Comment (pnarayanan) [inline ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java]: How is this going to be used? Based on the use case, you may not want to compare each field. For example, the `crc` might be optional in the `MessageInfo`. The fact that it is present in only one might not mean that the two `MessageInfo` instances represent different messages.

Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java]: It's to replace a MessageInfo comparison in a closed source transformer implementation test which currently just compares the toString outputs (which outputs a string concatenation of the values of all the fields that are being compared here).  I'm interested in equality of all fields in this scenario.

Comment (vgkholla) [reply ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java]: This is a good point - do you want to remove crc out of this or is that necessary?

Comment (dharju) [reply ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java]: I wanted an exact input == expected result so I'd like to have it in there for now

**Pull request merged**